bevy_lint/src/utils/hir_parse.rs

```diff
@@ -0,0 +1,44 @@
+//! Utility functions for parsing HIR types.
+
+use rustc_hir::{GenericArg, Node, QPath, Ty, TyKind};
+use rustc_lint::LateContext;
+
+/// Returns the list of types inside a tuple type.
+///
+/// If the type is not a tuple, returns a list containing the type itself.
+pub(crate) fn detuple(ty: Ty<'_>) -> Vec<Ty<'_>> {
+    if let TyKind::Tup(items) = ty.peel_refs().kind {
```

Comment:
But we can't pass the peeled type through since if it's not a tuple, we get the peeled type back instead of the original type. This means `(&Foo,)` returns `[&Foo]` but `&Foo` returns `[Foo]`.

If we don't peel the refs on the tuple here then the check has to move out to the call site anyways.

Maybe we add a `peel_tuple_refs: bool` parameter?

Or I can just remove the function if that's preferred.

Reply:
Oh, I understand. Thanks for the clarification! In this case, could you please document that this function peels references if it is a tuple?

Reply:
Sure thing!